src/T2DMap.cpp

```diff
@@ -672,13 +672,503 @@ bool T2DMap::sizeFontToFitTextInRect( QFont & font, const QRectF & boundaryRect,
     return true;
 }
 
+// This has been refactored to a separate function out of the paintEven() code
+// because we need to use it in two places - one for every room that is not the
+// player's room and then, AFTER all those have been drawn, once for the
+// player's room if it is visible. This is so it is drawn LAST (and any effects,
+// or extra markings for it do not get overwritten by the drawing of the other
+// rooms)...
+inline void T2DMap::drawRoom(QPainter& painter, QFont& roomVNumFont, QPen& pen, TRoom* pRoom, TArea* pArea, const int currentRoomId, const bool isFontBigEnoughToShowRoomVnum, const int playerRoomId, const float rx, const float ry, const bool picked)
```

Comment:
This function signature is a monster with many tentacles: I think if we gave anyone `drawRoom(painter, roomVNumFont, pen, room, pArea, currentAreaRoom, isFontBigEnoughToShowRoomVnum, playerRoomId, rx, ry, __Pick);` to look at, they'd have no clue what they were looking at. Let's improve it!

Can't `currentRoomId` be derived from `pRoom` inside the function? Or something like that - we have 3 arguments that all talk about the room, at least one should go.

`pArea` is only checked for `gridMode` - that should be passed in as a boolean to make the function interface more intuitive.

`isFontBigEnoughToShowRoomVnum` - no to yoda speak and **30** character variables! `roomTextfits` would work fine, observe:

```cpp
if (mShowRoomID && roomTextfits) {
```

And it wouldn't even need the comment it has right now, because it would be self-explanatory.

Reply:
The many tentacles come from me using the refactor code to extract a large chunk of code from the `paintEvent` code but not spending enough time to refine out ***all*** the extraneous bits - the use of `pArea` to get the `gridmode` setting does sound sensible, getting `currentAreaRoom` from `pRoom` also; `roomTextFits` though - what text - there is a *symbol* is *that* the text it means - fair enough I will have a think on that and find something we both can agree on...

Reply:
:worried: Oh, I haven't done much with this. Lemme have another look at it...

Reply:
... okay take a look at the latest commit: https://github.com/Mudlet/Mudlet/pull/2621/commits/010f8b4776bca4cb2e68eca78c2f9589859845f7 !